Audacia.Quest/Audacia.Quest.Core/Components/BaseComponent.cs

```diff
@@ -0,0 +1,102 @@
+﻿using Audacia.Quest.Core.Asset;
+
+namespace Audacia.Quest.Core.Components
+{
+    public abstract class BaseComponent : IComponent
+    {
+        public Guid Id { get; set; } = Guid.NewGuid();
+
+        public Transform Transform { get; set; } = new Transform();
+        public IComponent Parent { get; set; }
+        public List<ComponentMap> Components { get; set; } = new List<ComponentMap>();
+        public Sprite Sprite { get; set; }
+
+        private bool _enabled = true;
+
+        public bool Enabled
+        {
+            get
+            {
+                return _enabled;
+            }
+            set
+            {
+                var oldValue = _enabled;
+                _enabled = value;
+
+                if (value && oldValue != value)
```

Comment:
It depends on how your state works here but I'd say 1 of 2 things.
1. In the if statement use `_enabled` instead of `value`
2. move the statement for `_enabled = value;` to the end of the setter after you've finished initialising the component

Reply:
I'll do no. 1, but not 2 as I want to have `Enabled` be set by the time Init happens